x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/types.ts

```diff
@@ -162,5 +162,10 @@ export interface AlertAttributes {
 }
 
 export interface RuleAlertAttributes extends AlertAttributes {
-  params: Omit<RuleAlertParams, 'ruleId'> & { ruleId: string };
+  params: Omit<
+    RuleAlertParams,
+    'ruleId' | 'name' | 'enabled' | 'interval' | 'tags' | 'actions' | 'throttle'
+  > & {
+    ruleId: string;
+  };
```

Comment:
NOTE: This fixes the `params` types to remove the `interval` and other things that mislead programmers into thinking that the params contains those values when they are top level alerting options and not part of params.

Reply:
It looks like this type already exists (albeit without the `interval` change) as `RuleTypeParams`. Also, this looks VERY similar to [`RuleAlertType`](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/rules/types.ts#L41-L43) so I'm thinking these might be duplicated types

Reply:
These probably are. Really once we move over to io-ts we can get rid of a lot of these types. Hopefully very very soon.